src/components/explorer/ExplorerHeader.tsx

```diff
@@ -0,0 +1,333 @@
+import { buttonStyles } from 'components/common/button/buttonStyles';
+import { TokensOverlap } from 'components/common/tokensOverlap';
+import { useTokens } from 'hooks/useTokens';
+import { Strategy, useGetStrategyList } from 'libs/queries';
+import {
+  PairTrade,
+  Trending,
+  useTrending,
+} from 'libs/queries/extApi/tradeCount';
+import { Link } from 'libs/routing';
+import { CSSProperties, useEffect, useRef } from 'react';
+import { toPairSlug } from 'utils/pairSearch';
+
+const useTrendingPairs = (trending?: Trending) => {
+  const { tokensMap } = useTokens();
+  if (!trending) return { isLoading: true, data: [] };
+  const pairs: Record<string, PairTrade> = {};
+  for (const trade of trending?.pairCount ?? []) {
+    pairs[trade.pairAddresses] ||= trade;
+  }
+  const list = Object.values(pairs)
+    .filter((pair) => !!pair.pairTrades_24h)
+    .sort((a, b) => b.pairTrades - a.pairTrades)
+    .splice(0, 3);
+
+  // If there are less than 3, pick the remaining best
+  if (list.length < 3) {
+    const remaining = Object.values(pairs)
+      .filter((pair) => !!pair.pairTrades_24h)
+      .sort((a, b) => b.pairTrades - a.pairTrades)
+      .splice(0, 3 - list.length);
+    list.push(...remaining);
+  }
+
+  // Sort again in case we had to add more
+  const data = list
+    .sort((a, b) => b.pairTrades - a.pairTrades)
+    .map((pair) => ({
+      pairAddress: pair.pairAddresses,
+      base: tokensMap.get(pair.token0.toLowerCase())!,
+      quote: tokensMap.get(pair.token1.toLowerCase())!,
+      trades: pair.pairTrades,
+    }));
+  return { isLoading: false, data };
+};
+
+interface StrategyWithTradeCount extends Strategy {
+  trades: number;
+}
+const useTrendStrategies = (
+  trending?: Trending
+): { isLoading: boolean; data: StrategyWithTradeCount[] } => {
+  const trades = trending?.tradeCount ?? [];
```

Comment:
here too, the logic to extract IDs of trending strategies can be a standalone pure function - which is then passed to a query